swarm/src/behaviour.rs

```diff
@@ -387,6 +387,43 @@ impl<TOutEvent, THandlerIn> ToSwarm<TOutEvent, THandlerIn> {
             },
         }
     }
+
+    /// Map the event the swarm will return to an optional one.
+    pub fn map_out_opt<E>(
```

Comment:
Yeah indeed. It is just that we use it in our fork because we have several `Behaviour` wrapper and we often need to map an event into a different optional event. Currently there is no method like this on `ToSwarm`. I thought it was a good idea to upstream it. But if you prefer I can make it private in the `async_behaviour.rs` file.

Reply:
Removed since it is not used anymore with `bounded` channels.

It might be a nice thing to have though, but I will upstream it in a separate PR.